rdsbroker/broker.go

```diff
@@ -260,8 +260,13 @@ func (b *RDSBroker) restoreFromPointInTime(
 	provisionParameters ProvisionParameters,
 	servicePlan ServicePlan,
 ) error {
-	if servicePlan.RDSProperties.Engine != nil && *servicePlan.RDSProperties.Engine != "postgres" {
-		return fmt.Errorf("Restore from snapshot not supported for engine '%s'", *servicePlan.RDSProperties.Engine)
+	if servicePlan.RDSProperties.Engine != nil {
+		switch engine := *servicePlan.RDSProperties.Engine; engine {
+			case "postgres":
+			case "mysql":
+			default:
```

Comment:
This statement is confusing at first, and reads a bit like we don't support it for either. Consider

```
if servicePlan.RDSProperties.Engine != nil {
    if servicePlan.RDSProperties.Engine != "postgres" && servicePlan.RDSProperties.Engine != "mysql {
        ....
    }
}

```

Otherwise maybe we could look at adding some methods to the `ServicePlan` type? That could make some of the code a little easier to comprehend

```
func (sp *ServicePlan) SupportsPointInTimeRestore() bool {
    return sp.RDSProperties.Engine == "mysql" || sp.RDSProperties.Engine == "postgres"
}

...

func (b *RDSBroker) restoreFromPointInTime(
	ctx context.Context,
	instanceID string,
	details brokerapi.ProvisionDetails,
	asyncAllowed bool,
	provisionParameters ProvisionParameters,
	servicePlan ServicePlan,
) error {
    if !servicePlan.SupportsPointInTimeRestore() {
        return fmt.Errorf("Restore from snapshot not supported for engine '%s'", servicePlan.RDSPropertoes.Engine)
    } 
}
```

Reply:
Ok - you can tell I really just wanted to write `if sp.RDSProperties.Engine in ("mysql", "postgres")` but couldn't :(